commit 37afcd691f56807af68bc1c68ad6122a0a4f6cca
Author: Volker Krause <vkrause@kde.org>
Date:   Mon Jan 1 18:41:00 2018 +0100

    Make kdesingerplugin dependency optional
    
    As this involves a code generator that dynamically links against KF5 libs,
    this is a bit tricky to get right when cross-compiling. Fortunately, the
    designer plugin is only needed for development and thus is rather
    useless on the target platform anyway.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7826413d..16dc6242 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ find_package(KF5Completion ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Config ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5ConfigWidgets ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Crash ${KF5_DEP_VERSION} REQUIRED)
-find_package(KF5DesignerPlugin ${KF5_DEP_VERSION} REQUIRED)
+find_package(KF5DesignerPlugin ${KF5_DEP_VERSION})
 find_package(KF5GlobalAccel ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5DocTools ${KF5_DEP_VERSION})
 find_package(KF5Emoticons ${KF5_DEP_VERSION} REQUIRED)
diff --git a/KF5KDELibs4SupportConfig.cmake.in b/KF5KDELibs4SupportConfig.cmake.in
index cf3f2aca..99aef408 100644
--- a/KF5KDELibs4SupportConfig.cmake.in
+++ b/KF5KDELibs4SupportConfig.cmake.in
@@ -11,7 +11,6 @@ find_dependency(KF5Archive "@KF5_DEP_VERSION@")
 find_dependency(KF5ConfigWidgets "@KF5_DEP_VERSION@")
 find_dependency(KF5CoreAddons "@KF5_DEP_VERSION@")
 find_dependency(KF5Crash "@KF5_DEP_VERSION@")
-find_dependency(KF5DesignerPlugin "@KF5_DEP_VERSION@")
 if(@KF5DocTools_FOUND@)
     find_dependency(KF5DocTools "@KF5_DEP_VERSION@")
 endif()
diff --git a/cmake/modules/KDE4Macros.cmake b/cmake/modules/KDE4Macros.cmake
index 5c50a2cf..dc3c9af1 100644
--- a/cmake/modules/KDE4Macros.cmake
+++ b/cmake/modules/KDE4Macros.cmake
@@ -488,7 +488,9 @@ include("${CMAKE_CURRENT_LIST_DIR}/KDEUIMacros.cmake")
 if (KF5DocTools_FOUND)
     include("${KF5DocTools_DIR}/KF5DocToolsMacros.cmake")
 endif()
-include("${KF5DesignerPlugin_DIR}/KF5DesignerPluginMacros.cmake")
+if (KF5DesignerPlugin_FOUND)
+    include("${KF5DesignerPlugin_DIR}/KF5DesignerPluginMacros.cmake")
+endif()
 
 # functions deprecated, they still should work
 macro (KDE4_AUTH_INSTALL_ACTIONS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 469cc58a..df73c45b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -361,6 +361,7 @@ set_target_properties(KF5KDELibs4Support PROPERTIES
 
 ########## Designer plugin
 
+if (KF5DesignerPlugin_FOUND)
 kf5designerplugin_add_plugin(kf5deprecatedwidgets
     kdedeprecated.widgets
     kdedeprecated.qrc
@@ -377,6 +378,7 @@ target_link_libraries(kf5deprecatedwidgets KF5::KDELibs4Support
                                     )
 
 install(TARGETS kf5deprecatedwidgets DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/designer)
+endif()
 
 ##########
 
